### DISCUSSION TITLE
Can I check which subscriptions are active?

### DISCUSSION THREAD

**Discussion by cstrat** [mqttjs/MQTT.js#1768] [General]
This might be silly but I can't seem to find it in the API, how can I check to see which topics my client is currently subscribed to?

**Comment by robertsLando** [comment]
Actually this is pretty essy, I usually wrap the client inside another class to handle all main methods like publish/subscribe and connection management, in that case you simply keep track of calls to "subscribe"/unaubscribe and store the topics inside an array... alternatively you can check this: https://github.com/mqttjs/MQTT.js/issues/1762

Basically there is a private variable under client that is used to keep track of topics we need to resubscribe in case of a reconnection, you can access it even without a getter just use the square brackets instead of the point notation.

**Reply by cstrat** [reply]
Oh that private variable is probably what I will use.
I might find myself in a situation where I lose track of what subscriptions have been made. So to avoid double up subscriptions so I want to check the list before subbing again.
